### ISSUE TITLE
Exception in public_access_update when called in syncmodifyapi

### ISSUE THREAD

**Issue by mikkonie** [bihealth/sodar-server#1687]
As a part of working on #1617 I noticed a crash with a certain project, when performing `public_access_update` when called from `syncmodifyapi`. This may be related to recent changes within that issue, or something from before I simply didn't notice.

Full dump of error pasted below. I'll have to try and reproduce and see if this requires `syncmodifyapi` or can be triggered without it.

## Facts

- Public guest access is not enabled
- Sample sheets exist
- iRODS collections have not been created
    * I have a hunch this may be the root cause..

**Comment by mikkonie** [comment]
Log dump:

```
2023-05-31 11:00:29,602 [DEBUG] projectroles.management.commands.syncmodifyapi: Synchronizing project: Example Projects / Becnel Cohort
(Taskflow sync flows called here)
2023-05-31 11:00:30,015 [DEBUG] projectroles.views: Calling perform_project_sync() in plugin "samplesheets"
2023-05-31 11:00:30,083 [INFO] taskflowbackend.api: Building flow "public_access_update"..
2023-05-31 11:00:30,083 [INFO] taskflowbackend.api: Building flow OK
2023-05-31 11:00:30,083 [INFO] taskflowbackend.flows: Running flow "public_access_update"
2023-05-31 11:00:30,087 [INFO] taskflowbackend.tasks: Executed: <iRODS> Set collection user group access (SetAccessTask)
2023-05-31 11:00:30,089 [INFO] taskflowbackend.tasks: Reverted: <iRODS> Set collection user group access (SetAccessTask)
2023-05-31 11:00:30,090 [ERROR] taskflowbackend.flows: Exception in run_flow(): 
2023-05-31 11:00:30,091 [ERROR] taskflowbackend.api: Error running flow: 
2023-05-31 11:00:30,091 [ERROR] samplesheets.plugins: Public status update taskflow failed: Error running flow: 
2023-05-31 11:00:30,091 [ERROR] projectroles.views: Exception in perform_project_sync() for plugin "samplesheets": Error running flow: 
2023-05-31 11:00:30,091 [ERROR] projectroles.management.commands.syncmodifyapi: Exception in project sync: Error running flow: 
2023-05-31 11:00:30,091 [ERROR] projectroles.management.commands.syncmodifyapi: Project sync failed! Unable to continue, exiting..
```

**Comment by mikkonie** [comment]
When triggering public guest access from the project update menu, we get the following:

```
2023-05-31 12:06:59,212 [DEBUG] projectroles.views: Calling perform_project_modify() in plugin "samplesheets"
2023-05-31 12:06:59,221 [DEBUG] samplesheets.plugins: Skipping: Investigation collections not created in iRODS
```

That is correct. I guess the check is simply missing/broken in `perform_project_sync()`.

**Comment by mikkonie** [comment]
This turned out to be an existing bug. Fixed in the #1617 work branch, to be merged.
